src/core/support/slice_buffer.c

```diff
@@ -162,14 +163,30 @@ void gpr_slice_buffer_reset_and_unref(gpr_slice_buffer *sb) {
 }
 
 void gpr_slice_buffer_swap(gpr_slice_buffer *a, gpr_slice_buffer *b) {
-  gpr_slice_buffer temp = *a;
-  *a = *b;
-  *b = temp;
-
-  if (a->slices == b->inlined) {
+  GPR_SWAP(size_t, a->count, b->count);
+  GPR_SWAP(size_t, a->capacity, b->capacity);
+  GPR_SWAP(size_t, a->length, b->length);
+
+  if (a->slices == a->inlined) {
```

Comment:
if the three GPR_SWAP operations happened after the "if" (ie, as the last 3 operations of the function), the code would be less tricky. As it stands now, the {a,b}->counts are already swapped, which matters in reasoning about the memcpy size correctness.